third_party/move/move-compiler-v2/src/bytecode_generator.rs

```diff
@@ -1067,18 +1032,14 @@ impl<'env> Generator<'env> {
                     }
                 }
             },
-            ExpData::Call(id, Operation::MoveFunction(mid, fid), args) => {
-                // The type checker has ensured that this function returns a tuple
+            _ => {
```

Comment:
I believe this also fixes a bug I had filed a few weeks back: https://github.com/aptos-labs/aptos-core/issues/10716

It might be worth adding the test case mentioned there to our test.

Reply:
Added the test -- it passes.